From 34295889a65c360da159ffdb3c49bc7df3f75b17 Mon Sep 17 00:00:00 2001
From: Fabien Marotte <fabien.marotte@intel.com>
Date: Tue, 12 Jun 2012 10:49:24 +0200
Subject: [PATCH 09/78] widi: Change rcu synchronization to remove ADD_BEACON
 API delay

BZ: 38451

Changed code from synchronize_rcu() to synchronize_rcu_expedited()
This call change has no impact on the feature

Change-Id: Ib4ab2bff54b0daa6ee7166c2762d7d1c3497fe27
Signed-off-by: Fabien Marotte <fabien.marotte@intel.com>
Reviewed-on: http://android.intel.com:8080/52931
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Reviewed-by: Regairaz, Julie <julie.regairaz@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/59015
Reviewed-on: http://android.intel.com:8080/77966
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/net/mac80211/cfg.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/wl12xx-compat/net/mac80211/cfg.c b/wl12xx-compat/net/mac80211/cfg.c
index ddf2c3f..7d6f581 100644
--- a/wl12xx-compat/net/mac80211/cfg.c
+++ b/wl12xx-compat/net/mac80211/cfg.c
@@ -619,7 +619,7 @@ static int ieee80211_config_beacon(struct ieee80211_sub_if_data *sdata,
 
 	RCU_INIT_POINTER(sdata->u.ap.beacon, new);
 
-	synchronize_rcu();
+	synchronize_rcu_expedited();
 
 	kfree(old);
 
-- 
1.7.9

